src/Namicorn.ts

```diff
@@ -78,63 +79,112 @@ class Namicorn {
     }
   }
 
-/**
- * Resolves the given domain
- * @async
- * @param {string} domain - domain name to be resolved 
- * @returns {Promise<NamicornResolution>} - Returns a promise that resolves in an object 
- */
+  /**
+   * Resolves the given domain
+   * @async
+   * @param {string} domain - domain name to be resolved
+   * @returns {Promise<NamicornResolution>} - Returns a promise that resolves in an object
+   */
   async resolve(domain: string): Promise<NamicornResolution> {
-    if (this.blockchain) {
-      return await this.resolveUsingBlockchain(domain);
-    } else {
-      const response = isNode()
-        ? await fetch(`${this.api}/${domain}`)
-        : await window.fetch(`${this.api}/${domain}`);
-      return response.json();
+    try {
+      if (this.blockchain) {
+        return await this.resolveUsingBlockchain(domain);
+      } else {
+        const response = isNode()
+          ? await fetch(`${this.api}/${domain}`)
+          : await window.fetch(`${this.api}/${domain}`);
+        return response.json();
+      }
+    } catch (err) {
+      if (err.name == 'ResolutionError') {
```

Comment:
Instanceof doesn't catch the errors at all (the condition always returns false). I switched to err.name because of that and I set the name to 'ResolutionError' within the class.

Reply:
Hm that is weird. I used that many times in other places.